src/utils.zig

```diff
@@ -1137,110 +1158,145 @@ pub fn PaletteCompressedRegion(T: type, size: comptime_int) type { // MARK: Pale
 		}
 
 		pub fn getValue(self: *const Self, i: usize) T {
-			return self.palette[self.data.getValue(i)];
+			const impl = self.impl.load(.acquire);
+			return impl.palette[impl.data.getValue(i)].load(.unordered);
+		}
+
+		pub fn palette(self: *const Self) []Atomic(T) {
+			const impl = self.impl.raw;
+			return impl.palette[0..impl.paletteLength];
+		}
+
+		pub fn fillUniform(self: *Self, value: T) void {
+			const impl = self.impl.raw;
+			if(impl.paletteLength == 1) {
+				impl.palette[0].store(value, .unordered);
+				return;
+			}
+			var newSelf: Self = undefined;
+			newSelf.init();
+			newSelf.impl.raw.palette[0] = .init(value);
+			newSelf.impl.raw = self.impl.swap(newSelf.impl.raw, .release);
+			newSelf.deferredDeinit();
 		}
 
 		fn getOrInsertPaletteIndex(noalias self: *Self, val: T) u32 {
-			std.debug.assert(self.paletteLength <= self.palette.len);
+			var impl = self.impl.raw;
+			std.debug.assert(impl.paletteLength <= impl.palette.len);
 			var paletteIndex: u32 = 0;
-			while(paletteIndex < self.paletteLength) : (paletteIndex += 1) { // TODO: There got to be a faster way to do this. Either using SIMD or using a cache or hashmap.
-				if(std.meta.eql(self.palette[paletteIndex], val)) {
+			while(paletteIndex < impl.paletteLength) : (paletteIndex += 1) { // TODO: There got to be a faster way to do this. Either using SIMD or using a cache or hashmap.
```

Comment:
Since we already touched this line, can we please avoid leaving TODOs in code? I already found few outdated ones, so I would suggest creating an issue and at least leaving an issue link or even better removing the TODO note completely. With VSC with Github extensions (not sure which one, maybe PR one) is capable of showing issues if one leaves `#<issue-number>` in TODO comment.

Reply:
#318